core/src/modules/ops/destination.ts

```diff
@@ -912,82 +923,109 @@ export namespace DestinationOps {
     success: boolean;
     error: Error;
   }> {
-    const _exports: Export[] = []; // only ones we are sending
-    for (const _export of givenExports) {
-      // TODO: maybe should recalcuate hasChanges and from current mostRecent
-      if (!_export.hasChanges) {
-        await _export.complete(); // do not include exports with hasChanges=false
-      } else {
-        _exports.push(_export);
-      }
-    }
+    const _exports: Export[] = [];
+    const locks: Awaited<ReturnType<typeof getLock>>[] = [];
 
-    const exportRecords: ExportRecordsPluginMethod = await getBatchFunction(
-      destination
-    );
+    try {
+      for (const givenExport of givenExports) {
+        if (!givenExport.hasChanges) await givenExport.complete();
 
-    const syncMode = await destination.getSyncMode();
-    const syncOperations: DestinationSyncOperations = syncMode
-      ? DestinationSyncModeData[syncMode].operations
-      : DestinationSyncModeData.sync.operations; // if destination does not support sync modes, allow all
+        const lock = await getLock(
+          `${givenExport.recordId}:${givenExport.destinationId}`
+        );
 
-    const options = await destination.getOptions();
-    const app = await destination.$get("app", {
-      scope: null,
-      include: [Option],
-    });
-    const appOptions = await app.getOptions();
-    const connection = await app.getConnection();
+        const mostRecentExport = await Export.findOne({
+          where: {
+            recordId: givenExport.recordId,
+            destinationId: destination.id,
+          },
+          order: [["createdAt", "DESC"]],
+        });
+
+        const isNewest = mostRecentExport.id === givenExport.id;
+        if (!isNewest) console.info(mostRecentExport);
+        const gotLock = typeof lock === "function";
```

Comment:
what else can it be?

Reply:
`null`

Reply:
I meant `typeof` - I was just thinking we would use `!lock`

Reply:
For sure.  @evantahler had recommended this, I believe as a slightly tighter guarantee.

Reply:
As you like. I think it's even weirder if it somehow returns something else and we don't unlock it. I'd rather see the crash in `Promise.all` below if it's not a function somehow. But not strong feelings here.